### PR TITLE
feat(primitives): move `revm-primitives` to `reth-primitives`

### PR DESCRIPTION
**Pull request by tcoratger**
Should be part of #4714.

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
hmm, this moves everything from revm-primtiives to reth-primitives.

I think we can do that, which would make it easier to access all these helper/compat functions directly from reth-primitives.

wdyt @rakita

**Review by rakita — APPROVED** [review]
lgtm!

File: crates/storage/codecs/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ value-256 = ["codecs-derive/value-256"]
 bytes.workspace = true
 codecs-derive = { path = "./derive", default-features = false }
 revm-primitives = { workspace = true, features = ["serde"] }
+alloy-primitives = { workspace = true, features = ["rand", "rlp"] }
```
Comment (mattsse) [inline crates/storage/codecs/Cargo.toml:23]: I don't like this, we should stick to reexported types for internal crates

File: crates/storage/codecs/src/lib.rs [hunk 2]
```diff
@@ -8,11 +8,8 @@
 
 pub use codecs_derive::*;
 
+use alloy_primitives::{Address, Bloom, Bytes, B256, B512, U256};
```
Comment (mattsse) [inline crates/storage/codecs/src/lib.rs:11]: see comment above

**Review by mattsse — CHANGES_REQUESTED** [review]
okay, convinced.

smol nit

File: crates/storage/codecs/src/lib.rs (hunk 2 above)
Comment (tcoratger) [reply crates/storage/codecs/src/lib.rs]: @mattsse Done

File: crates/storage/codecs/Cargo.toml (hunk 1 above)
Comment (tcoratger) [reply crates/storage/codecs/Cargo.toml]: @mattsse Done

**Review by mattsse — APPROVED** [review]
lgtm

maybe we need to rethink the reth-revm crate now, perhaps we can find a more appropriate name for it, it also doesn't need to depend on the inspectors crate I believe

**Pull request merged**